### PR TITLE
[FIX] account: Block archival of journals with auto post entries

### PR DESCRIPTION
**Pull request by vin-odoo**
At the moment, it is possible to archive a journal for which some
entries are still waiting to be auto-posted in the future.
Add a constrains to avoid make sure that it is no longer possible,
since it shouldn't.

Task id #2585454

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_journal.py [hunk 1]
```diff
@@ -323,6 +323,21 @@ def _check_type_default_account_id_type(self):
             if journal.type in ('sale', 'purchase') and journal.default_account_id.user_type_id.type in ('receivable', 'payable'):
                 raise ValidationError(_("The type of the journal's default credit/debit account shouldn't be 'receivable' or 'payable'."))
 
+    @api.constrains('active')
+    def _check_auto_post_draft_entries(self):
+        for journal in self:
+            pending_moves = self.env['account.move'].search([
+                ('journal_id', '=', journal.id),
+                ('state', '=', 'draft')
+            ])
```
Comment (smetl) [inline addons/account/models/account_journal.py:332]: add `limit=1` to not force the orm to fetch everything for nothing ;)

**Review by smetl — CHANGES_REQUESTED** [review]

**Comment by smetl** [comment]
Also, add link to the task into the commit message

File: addons/account/models/account_journal.py (hunk 1 above)
Comment (vin-odoo) [reply addons/account/models/account_journal.py]: Yep indeed sorry about that

**Review by smetl — APPROVED** [review]
@robodoo r+

File: addons/account/models/account_journal.py (hunk 1 above)
Comment (sswapnesh) [reply addons/account/models/account_journal.py]: or just `search_count`

**Pull request closed without merging**